constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/helpers/HorizontalChainReference.java

```diff
@@ -51,21 +51,31 @@ public void apply() {
                     first.startToEnd(mStartToEnd).margin(mMarginStart).marginGone(mMarginStartGone);
                 } else if (mLeftToLeft != null) {
                     // TODO: Hack until we support RTL properly
-                    first.startToStart(mLeftToLeft).margin(mMarginLeft).marginGone(mMarginLeftGone);
+                    first.leftToLeft(mLeftToLeft).margin(mMarginLeft).marginGone(mMarginLeftGone);
                 } else if (mLeftToRight != null) {
                     // TODO: Hack until we support RTL properly
-                    first.startToEnd(mLeftToRight).margin(mMarginLeft).marginGone(mMarginLeftGone);
+                    first.leftToRight(mLeftToRight).margin(mMarginLeft).marginGone(mMarginLeftGone);
                 } else {
                     // No constraint declared, default to Parent.
                     String refKey = reference.getKey().toString();
-                    first.startToStart(State.PARENT).margin(getPreMargin(refKey));
+                    if (lrMode) {
+                        first.leftToLeft(State.PARENT).margin(getPreMargin(refKey));
+
+                    } else {
+                        first.startToStart(State.PARENT).margin(getPreMargin(refKey));
+                    }
                 }
             }
             if (previous != null) {
                 String preKey = previous.getKey().toString();
                 String refKey = reference.getKey().toString();
-                previous.endToStart(reference.getKey()).margin(getPostMargin(preKey));
-                reference.startToEnd(previous.getKey()).margin(getPreMargin(refKey));
+                if (lrMode) {
+                    previous.rightToLeft(reference.getKey()).margin(getPostMargin(preKey));
+                    reference.leftToRight(previous.getKey()).margin(getPreMargin(refKey));
+                } else {
+                    previous.endToStart(reference.getKey()).margin(getPostMargin(preKey));
+                    reference.startToEnd(previous.getKey()).margin(getPreMargin(refKey));
+                }
```

Comment:
I suppose this isn't wrong, but as it is, it makes no difference.

`startToStart` will always apply `leftToLeft` constraints.

Reply:
When we implement left,right,start and end this is the correct behavior